### PR TITLE
Space Item Pinning

### PR DESCRIPTION
**Pull request by buracules**
- Added space item pinning capability

### How was this tested?
- [x] I tested this manually
- [x] I added automated tests
- [ ] Existing automated tests are enough to cover my changes

### Screenshots and screen recordings

https://user-images.githubusercontent.com/15333612/180238862-aa4fb89c-9305-4e77-9465-04f4effa2be0.mp4

### CODE REVIEW

File: Client/Frontend/Browser/Spaces/SpaceDetailList.swift [hunk 1]
```diff
@@ -195,9 +221,46 @@ struct SpaceDetailList: View {
...
+        }
+
+        primitive.item?.contentData?[index].isPinned = isPinned
+        primitive.allDetails[index].isPinned = isPinned
+
+        if isPinned,
+            let indexInSection = primitive.unpinnedDetails.firstIndex(where: {
+                $0.item?.id == spaceItemId
+            }),
+            let indexInDetails = primitive.allDetails.filter({ $0.isPinned }).firstIndex(
```
Comment (buracules) [inline Client/Frontend/Browser/Spaces/SpaceDetailList.swift:237]: `indexInDetails` is for putting back the item to where it was.

**Comment by evancrow** [comment]
@buracules Is this still a WIP?

**Comment by buracules** [comment]
> @buracules Is this still a WIP?

We left this one unfinished i don't remember why

**Comment by buracules** [comment]
Marked as draft for now fyi @evancrow

**Comment by jedireza** [comment]
Closing this due to it's age. Feel free to re-open if it's still an active work stream.

**Pull request closed without merging**